lib/piwik/index.js

```diff
@@ -64,8 +67,42 @@ Piwik.prototype.track = function(track){
   var name = track.proxy('properties.name') || track.proxy('properties.label');
   var value = track.value() || track.revenue();
 
+  var options = track.options('Piwik');
+  var customVariableLimit = this.options.customVariableLimit || 5;
+  var customVariables = options.customVars || options.cvar;
+
+  if (!is.object(customVariables) && !is.array(customVariables)) {
+    customVariables = [];
+  }
+
+  if (is.object(customVariables)) {
+    customVariables = this._formatToArray(customVariables);
+  }
+
+  for (var i = 0; i < customVariableLimit; i += 1){
+    if (customVariables[i]) {
+      var varIndex = (i + 1).toString();
+      push('setCustomVariable', varIndex, customVariables[i][0], customVariables[i][1], 'page');
+    }
+  }
+
   each(goals, function(goal){
     push('trackGoal', goal, revenue);
   });
+
   push('trackEvent', category, action, name, value);
 };
+
+/**
+ * Helper method to format the custom variables.
+ *
+ * @param {Object} object
+ */
+
+Piwik.prototype._formatToArray = function(object){
+  return reduce(object, [], function(result, key, value){
+    key = parseInt(key, 10);
```

Comment:
We aren't using the key here, just parsing it. Also, we aren't guaranteed to store the value at the index given by `key` because order of iteration over objects isn't guaranteed as part of the ES spec*, so you need to parse and use the key here. It also won't filter out non-numeric keys. Check out my prior comment for the fix.

Run this code for a repro:

```js
var obj = {};
obj.a = 'shirt'
obj['1'] = 'pants';
obj['0'] = 'shoes';
for (var key in obj) { console.log(key); }
```

* Iteration order behavior is platform specific: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in